src/docs/BUILD.md

```diff
@@ -3,73 +3,149 @@
 > **NOTE**: There is a known Python API issue with `ov::Tensor`. The issue is reproduced when building OpenVINO GenAI from sources while using OpenVINO from archives. Using `ov::Tensor` with OpenVINO GenAI fails. Possible errors: `TypeError: generate(): incompatible function arguments.`, `TypeError: __init__(): incompatible constructor arguments.`, `TypeError: Unregistered type : ov::Tensor`.
 The preferred approach is to build both OpenVINO and OpenVINO GenAI from sources using the same build environment. Or to install prebuilt OpenVINO GenAI from [distribution channels](https://docs.openvino.ai/2024/get-started/install-openvino.html).
 
-## Build for Linux Systems
+## Software Requirements
 
-### Software Requirements 
+### Linux
 
 - [CMake](https://cmake.org/download/) 3.23 or higher
 - GCC 7.5 or higher
 - Python 3.8 or higher
+- Git
 
-### Build Instructions
+### Windows
+
+- [CMake](https://cmake.org/download/) 3.23 or higher
+- Microsoft Visual Studio 2019 or higher, version 16.3 or later
+- Python 3.8 or higher
+- Git for Windows
+- [NSIS](https://sourceforge.net/projects/nsis/)
+
+### macOS
+
+- [CMake](https://cmake.org/download/) 3.23 or higher
+- [brew](https://brew.sh/) package manager to install additional dependencies:
+    ```sh
+    brew install coreutils scons
+    ```
+- Clang compiler and other command line tools from Xcode 10.1 or higher:
+    ```sh
+    xcode-select --install
+    ```
+- Python 3.8 or higher
+- Git
+
+
+## Build Instructions
+
+### Build OpenVINO, OpenVINO Tokenizers, and OpenVINO GenAI From Source
 
 1. Build and install OpenVINO from sources following the [instructions](https://github.com/openvinotoolkit/openvino/wiki#how-to-build).  
-The path to the openvino install directory is referred as <INSTALL_DIR> throughout the document.
+The path to the OpenVINO install directory is referred as `<INSTALL_DIR>` throughout the document.
 2. Clone OpenVINO GenAI repository and init submodules:
     ```sh
     git clone --recursive https://github.com/openvinotoolkit/openvino.genai.git
     cd openvino.genai
     ```
-3. Build the project:
+3. Set up the environment:
+
+    #### Option 1 - using OpenVINO `setupvars` script:
+
+    Linux and macOS:
     ```sh
     source <INSTALL_DIR>/setupvars.sh
-    cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-    cmake --build ./build/ --config Release --target package -j
-    cmake --install ./build/ --config Release --prefix ov
     ```
 
-## Build for Windows Systems
+    Windows Command Prompt:
+    ```cmd
+    call <INSTALL_DIR>\setupvars.bat
+    ```
 
-### Software Requirements 
+    Windows PowerShell:
+    ```cmd
+    . <INSTALL_DIR>/setupvars.ps1
+    ```
 
-- [CMake](https://cmake.org/download/) 3.23 or higher
-- Microsoft Visual Studio 2019 or higher, version 16.3 or later
-- Python 3.8 or higher
-- Git for Windows
+    #### Option 2 - setting environment variables manually:
```

Comment:
why do we need to recommend such approach? setupvars is default and we don't need to expose variables manually (which is implementation detail), because we will have to support it in future in case of any changes in OpenVINO scripts.

Reply:
Developers tend to stop at build step and skip installation part. @TolyaTalamanov is an example.

Reply:
it's a developer problems that they cannot read readme step by step

Reply:
They can. But they aren't explained that it's not recommended to skip it and why.